src/app/hooks/quota.ts

```diff
@@ -30,39 +30,42 @@ export const useQuota = (productId: ProductType) => {
     }, [config?.ams.apiBasePath, auth]);
 
     const getQuotaTypesByProductId = () => {
-        const kasQuotaId = constants.kafka.ams.instanceQuotaId
-        const kasTrialQuotaId  =  constants.kafka.ams.trialQuotaId
-        const srsQuotaId = constants.serviceRegistry.ams.instanceQuotaId
-        const srsTrialQuotaId  =  constants.serviceRegistry.ams.trialQuotaId
+        const kasQuotaProductId = constants.kafka.ams.quotaProductId;
+        const kasTrialQuotaProductId = constants.kafka.ams.trialQuotaProductId;
+        const kasResourceName = constants.kafka.ams.resourceName;
+
+        const srsQuotaProductId = constants.serviceRegistry.ams.quotaProductId;
+        const srsTrialQuotaProductId = constants.serviceRegistry.ams.trialQuotaProductId;
+        const srsResourceName = constants.serviceRegistry.ams.resourceName;
+
         if (productId === ProductType.kas) {
-            return { quotaId: kasQuotaId, trialQuotaId: kasTrialQuotaId, quotaKey: QuotaType.kas, trialQuotaKey: QuotaType.kasTrial };
+            return { quotaProductId: kasQuotaProductId, trialQuotaProductId: kasTrialQuotaProductId, resourceName: kasResourceName, quotaKey: QuotaType.kas, trialQuotaKey: QuotaType.kasTrial };
         } else if (productId === ProductType.srs) {
-            return { quotaId: srsQuotaId, trialQuotaId: srsTrialQuotaId, quotaKey: QuotaType.srs, trialQuotaKey: QuotaType.srsTrial };
+            return { quotaProductId: srsQuotaProductId, trialQuotaProductId: srsTrialQuotaProductId, resourceName: srsResourceName, quotaKey: QuotaType.srs, trialQuotaKey: QuotaType.srsTrial };
         }
     }
 
     const getQuota = async () => {
         let filteredQuota: Quota = { loading: true, isServiceDown: false, data: undefined };
 
         if (orgId) {
-            const { quotaId, trialQuotaId, quotaKey, trialQuotaKey } = getQuotaTypesByProductId() || {};
+            const { quotaProductId, trialQuotaProductId, resourceName, quotaKey, trialQuotaKey } = getQuotaTypesByProductId() || {};
 
             const accessToken = await auth?.ams.getToken();
             const ams = new AppServicesApi({
                 accessToken,
                 basePath: config?.ams.apiBasePath || '',
             } as Configuration);
-
             await ams
                 .apiAccountsMgmtV1OrganizationsOrgIdQuotaCostGet(orgId, undefined, true)
                 .then((res) => {
```

Comment:
await ams...then: is that correct?

Reply:
yes, It's working fine. Here we have mix of await and then. I'll update it. Thanks for the suggestion.

Reply:
yeah, it works, but it converts the `non async` method to e `resolved promise`, causing some overhead. See [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await#description)